data/csv.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+import (
+	"encoding/csv"
+	"log"
+	"os"
+)
+
+type csvDataSource struct {
+	// Collection is a file path to a CSV file
+	Collection string
+}
+
+func NewCSVDataSource(csvPath string) csvDataSource {
+	return csvDataSource{Collection: csvPath}
+}
+
+func (ds csvDataSource) ReadCollection() [][]string {
+	file, err := os.Open(ds.Collection)
+	if err != nil {
```

Comment:
Pro-tip, you can check for the error at the same line you declared it as following:
```suggestion
	if file, err := os.Open(ds.Collection); err != nil {
```

Reply:
But if I do this, wouldn't the `file` variable become scoped to just the `if` block? Making it invisible for the rest of the function code.

Reply:
Oh, yes, you are right, I didn't see that. Please omit that pro-tip (you can use this trick in other scenario though).